.github/workflows/build-pr.yml

```diff
@@ -19,3 +19,45 @@ jobs:
         run: make
       - name: Check build
         run: bin/ratify version
+
+  build_test_e2e:
+    name: "[External Data] Build and run e2e Test"
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    permissions:
+      contents: read
+    strategy:
+      matrix:
+        KUBERNETES_VERSION: ["1.23.0"]
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: Set up Go 1.17
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.17
+
+      - name: Bootstrap e2e
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/bin
+          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+          make e2e-bootstrap KUBERNETES_VERSION=${{ matrix.KUBERNETES_VERSION }}
+      - name: Run e2e
+        run: |
+          make e2e-deploy-gatekeeper
+          make e2e-deploy-ratify
+          make test-e2e
+      - name: Save logs
+        if: ${{ always() }}
+        run: |
+          kubectl logs -n gatekeeper-system -l control-plane=controller-manager --tail=-1 > logs-externaldata-controller.json
+          kubectl logs -n gatekeeper-system -l control-plane=audit-controller --tail=-1 > logs-externaldata-audit.json
+          kubectl logs -n ratify-service -l app.kubernetes.io/name=ratify --tail=-1 > logs-ratify.json
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
```

Comment:
Just curious, where do these logs get uploaded to and what are they used for?

Reply:
the logs can be downloaded from the run link https://github.com/deislabs/ratify/actions/runs/2316470812, artifact section